pysr/sr.py

```diff
@@ -337,25 +334,13 @@ def init_julia():
     return Main
 
 
-def _write_project_file(tmp_dir):
-    """This writes a Julia Project.toml to a temporary directory
-
-    The reason we need this is because sometimes Python will compile a project to binary,
-    and then Julia can't read the Project.toml file. It is more reliable to have Python
-    simply create the Project.toml from scratch.
-    """
-
-    project_toml = """
-[deps]
-SymbolicRegression = "8254be44-1295-4e6a-a16d-46603ac705cb"
-
-[compat]
-SymbolicRegression = "0.7.7, 0.7.8"
-julia = "1.5"
-    """
-
-    project_toml_path = tmp_dir / "Project.toml"
-    project_toml_path.write_text(project_toml)
+def _add_sr_to_julia_project(Main, io_arg):
+    Main.spec = Main.PackageSpec(
+        name="SymbolicRegression",
+        url="https://github.com/MilesCranmer/SymbolicRegression.jl",
+        rev="v0.7.8",
+    )
+    Main.eval(f"Pkg.add(spec, {io_arg})")
```

Comment:
This confuses me. What is this second `io_arg` argument?
https://pkgdocs.julialang.org/v1/api/#Pkg.add

Reply:
Ah, I see https://pkgdocs.julialang.org/v1/api/#Redirecting-output

Reply:
Just to add: the reason I have to do the `Main.eval` here is because the PyObject wrapper for IOStreams doesn't seem to pass correctly from Python. It's a bit ugly but works for now. The reason I pass the full `io_arg` instead of something like `io={io}` is because the IO redirection was only implemented in Julia 1.5+.